### PR TITLE
Part of 17670: Replace Typography with Text in nft-details.js

### PR DESCRIPTION
**Pull request by VedanthB**
## Explanation

Updating Typography to Text `nft-details.js`

Part of: #17670

### Before

beforee

### After

after

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by NidhiKJha — COMMENTED** [review]
Thanks for this contribution @VedanthB. Really great start. The `text` component is not using boxProps, please update the margin prop. Also, it would be good, if you could attach a screenshot for the change.

File: ui/components/app/nft-details/nft-details.js [hunk 1]
```diff
@@ -360,9 +370,10 @@ export default function NftDetails({ nft }) {
                 className="nft-details__link-title"
               >
                 {t('link')}
-              </Typography>
-              <Typography
-                variant={TypographyVariant.H6}
+              </Text>
+              <Text
+                variant={TextVariant.bodySm}
```
Comment (georgewrmarshall) [inline ui/components/app/nft-details/nft-details.js:375]: Change 
```
                variant={TextVariant.bodySm}
                as="h6"
                boxProps={{
                  margin: 0,
                  marginBottom: 4,
                }}
```
to
```
                variant={TextVariant.bodySm}
                as="h6"
                marginBottom={4}
```

File: ui/components/app/nft-details/nft-details.js [hunk 2]
```diff
@@ -380,13 +391,14 @@ export default function NftDetails({ nft }) {
...
-              </Typography>
+              </Text>
             </Box>
           ) : null}
           <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.ROW}>
-            <Typography
+            <Text
               color={TextColor.textDefault}
-              variant={TypographyVariant.H6}
+              variant={TextVariant.bodySm}
```
Comment (georgewrmarshall) [inline ui/components/app/nft-details/nft-details.js:400]: Change
```
              variant={TextVariant.bodySm}
              as="h6"
              fontWeight={FONT_WEIGHT.BOLD}
              boxProps={{
                margin: 0,
                marginBottom: 4,
                marginRight: 2,
              }}
```
to
```
              variant={TextVariant.bodySmBold}
              as="h6"
              marginBottom={4}
              marginRight={2}
```

File: ui/components/app/nft-details/nft-details.js [hunk 3]
```diff
@@ -396,20 +408,21 @@ export default function NftDetails({ nft }) {
...
               className="nft-details__contract-wrapper"
             >
-              <Typography
+              <Text
                 color={TextColor.textAlternative}
-                variant={TypographyVariant.H6}
+                variant={TextVariant.bodySm}
+                as="h6"
                 overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
                 boxProps={{ margin: 0, marginBottom: 4 }}
```
Comment (georgewrmarshall) [inline ui/components/app/nft-details/nft-details.js:422]: Change
```
                boxProps={{ margin: 0, marginBottom: 4 }}
```
to
```
                marginBottom={4}
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Great start! Left some suggestions that weren't listed in the issue. Would be great to get some before/after screenshots to ensure that there is no visual differences

**Comment by VedanthB** [comment]
@georgewrmarshall , @NidhiKJha I have made the changes please review.

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Just need to update snapshots

```
yarn jest ui/components/app/nft-details --updateSnapshot
```

**Comment by VedanthB** [comment]
> Looking good! Just need to update snapshots
> 
> ```
> yarn jest ui/components/app/nft-details --updateSnapshot
> ```

@georgewrmarshall  Done, please review.

**Comment by NidhiKJha** [comment]
@VedanthB Can you please resolve the conflicts for the build to pass

**Review by georgewrmarshall — COMMENTED** [review]
This should be the correct import to resolve your conflict 

```
import { ButtonIcon, IconName, Text } from '../../component-library';
```

**Review by georgewrmarshall — COMMENTED** [review]
LGTM 🚀 Just need tests to pass and I will approve

File: ui/components/app/nft-details/nft-details.js [hunk 4]
```diff
@@ -202,42 +200,44 @@ export default function NftDetails({ nft }) {
...
             <div>
-              <Typography
+              <Text
                 color={TextColor.textDefault}
-                variant={TypographyVariant.H4}
-                fontWeight={FONT_WEIGHT.BOLD}
-                boxProps={{ margin: 0, marginBottom: 2 }}
+                variant={TextVariant.headingSm}
+                as="h4"
+                marginBottom={2}
```
Comment (DDDDDanica) [inline ui/components/app/nft-details/nft-details.js:207]:
```suggestion
                marginBottom={2}
                marginTop={0}
                marginLeft={0}
                marginRight={0}
```

File: ui/components/app/nft-details/nft-details.js [hunk 5]
```diff
@@ -202,42 +200,44 @@ export default function NftDetails({ nft }) {
...
+              </Text>
             </div>
             {description ? (
               <div>
-                <Typography
+                <Text
                   color={TextColor.textDefault}
-                  variant={TypographyVariant.H6}
-                  fontWeight={FONT_WEIGHT.BOLD}
+                  variant={TextVariant.bodySmBold}
```
Comment (DDDDDanica) [inline ui/components/app/nft-details/nft-details.js:225]:
```suggestion
                  variant={TextVariant.bodySm}
```

File: ui/components/app/nft-details/nft-details.js (hunk 4 above)
Comment (DDDDDanica) [reply ui/components/app/nft-details/nft-details.js:207]: @georgewrmarshall I'm curious if there's a default value when we don't pass in any for margin, maybe i'm wrong in this one.

Comment (georgewrmarshall) [reply ui/components/app/nft-details/nft-details.js:207]: So the `Typography` component had some default margin values set which in this instance were being removed via the `boxProps` the `Text` component doesn't have any default margins set so they are already the equivalent of `0`

Comment (DDDDDanica) [reply ui/components/app/nft-details/nft-details.js:207]: Gotcha. Thanks for explaining. Will resolve this since it’s not valid.

**Comment by georgewrmarshall** [comment]
Hey @VedanthB, are you able to resolve the conflicts and we can get this merged. The import statement should look like the below

```
`import { ButtonIcon, IconName, Text } from '../../component-library';`
```

**Comment by georgewrmarshall** [comment]
Hey @VedanthB, it looks like another PR was merged that migrated the `Text` components for these files. Thanks very much for your contribution and I encourage you to contribute in future.

**Pull request closed without merging**